# Archaeome

Recent studies described the enteric microbiomes of fish by amplicon sequencing using universal primers which miss much of the archaeal diversity present. The previous studies performed on the intestinal tracts of terrestrial vertebrates suggest that methanogens represent the major part of the archaeome and might be a considerable source of methane emission into the atmosphere. Fish represent the highest biomass of vertebrates on earth and we still do not know their potential contribution in methane emission. Using 16S rRNA amplicon sequencing with 13 Archaea-targeted primers and quantitative PCR on methanogens with the mcrA gene, we showed that Archaea and specifically methanogens are commonly absent (7,8% amplification success) or exceptionally rare (> 100 copy/ÂµL). The available prokaryotic 16S rRNA sequences provided from the same samples, indicated the same trend with 0.05% of Archaea, mainly constituted by non-methanogens Thaumarchaeaota. We measured the emission of gas produced by Mediterranean fish in vivo by gas chromatography but we did not detect any methane production, suggesting that marine fish represent a methane-free source of proteins. In a second time, we tested the presence of sulfate-reducing bacteria (SRB) targeting the dsrB gene with qPCR using specific primers and with the prokaryotic 16S rRNA amplicon sequences. The success of amplification of the dsrB gene (37%) was very variable between sample fish and goes together with the result of the Tax4Fun2 prediction on the genes involved in sulfate-reducing metabolisms with a global contribution of 0.02% of the total functional gene contribution. As such, sulfate-reduction may be favored in the gut microbiome of marine fish and compete with methanogens, thus limiting the methane emission.
